### PR TITLE
Add routes for add/delete entity to the catalog

### PR DESCRIPTION
**Pull request by freben**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by freben** [comment]
Hang on, adding some more tests and structure to this ...

File: plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.test.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+ * limitations under the License.
+ */
+
+import type { MockedMemberFunctions } from '@backstage/backend-common';
+import type { Entity, EntityPolicy } from '@backstage/catalog-model';
+import type { Database } from '../database';
+import { DatabaseEntitiesCatalog } from './DatabaseEntitiesCatalog';
+
+describe('DatabaseEntitiesCatalog', () => {
+  let db: MockedMemberFunctions<Database>;
```
Comment (Rugvip) [inline plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.test.ts:23]:
```suggestion
  let db: jest.Mocked<Database>;
```

File: packages/backend-common/src/testing/MockedMemberFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * you need to write expect(x.y as jest.Mock).toHaveBeenCalled... because the
+ * x.y member was considered to be the actual function type. You could also
+ * change your test to instead create a "raw" object { y: jest.fn() } but then
+ * you lose type safety when doing x.y.mockReturnValue(...). So you start
+ * trying to do { y: jest.fn() as X['y'] } as X or similar trickery.
+ *
+ * This type lets you say const x: MockedMemberFunctions<X> = { y: jest.fn() }
+ * and keep all the type safety at every step.
+ */
+export type MockedMemberFunctions<T> = {
```
Comment (Rugvip) [inline packages/backend-common/src/testing/MockedMemberFunctions.ts:31]: Can use the builtin `jest.Mocked` instead

File: plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts [hunk 3]
```diff
@@ -14,12 +14,15 @@
  * limitations under the License.
  */
 
-import { Entity } from '@backstage/catalog-model';
-import { Database } from '../database';
-import { EntitiesCatalog, EntityFilters } from './types';
+import type { Entity, EntityPolicy } from '@backstage/catalog-model';
+import type { Database, DbEntityResponse, EntityFilters } from '../database';
+import type { EntitiesCatalog } from './types';
```
Comment (Rugvip) [inline plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts:19]: I don't think we actually need to use type-only imports, but it doesn't hurt :>

File: plugins/catalog-backend/src/database/types.ts [hunk 4]
```diff
@@ -83,3 +83,82 @@ export type DatabaseLocationUpdateLogEvent = {
...
+export type Database = {
+  /**
+   * Runs a transaction.
+   *
+   * The callback is expected to make calls back into this class. When it
+   * completes, the transaction is closed.
+   *
+   * @param fn The callback that implements the transaction
+   */
+  transaction<T>(fn: (tx: unknown) => Promise<T>): Promise<T>;
```
Comment (Rugvip) [inline plugins/catalog-backend/src/database/types.ts:106]: could add a type parameter to `Database` for the tx type right?

**Review by Rugvip — APPROVED** [review]
Yay for DB interface

File: packages/backend-common/src/testing/MockedMemberFunctions.ts (hunk 2 above)
Comment (freben) [reply packages/backend-common/src/testing/MockedMemberFunctions.ts:31]: Whoa I went through all this for nothing?!

File: plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts (hunk 3 above)
Comment (freben) [reply plugins/catalog-backend/src/catalog/DatabaseEntitiesCatalog.ts:19]: yeah it kinda became a habit of sorts... at least when i remember :)

File: plugins/catalog-backend/src/database/types.ts (hunk 4 above)
Comment (freben) [reply plugins/catalog-backend/src/database/types.ts:106]: Yes, that would work. I wanted to kinda explicitly make it opaque though, so the outside can't see it at all (essentially saying, by ways of API design, "don't you bother your head about this")

Comment (Rugvip) [reply plugins/catalog-backend/src/database/types.ts:106]: I think that would be achieved with a type parameter as well? Some `Database<Tx = unknown>` kinda thing

**Pull request merged**